crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -0,0 +1,189 @@
+                    Some(value)
+                } else {
+                    None
+                }
+            })
+        }
+        .ok_or_else(|| Error::Internal(format!("Field '{}' not found", field_name)).extend()),
+        _ => Err(Error::Internal("Unexpected Move struct type".to_string()).extend()),
+    }
+}
+
+#[Object]
+impl DynamicField {
+    async fn name(&self, ctx: &Context<'_>) -> Result<Option<MoveValue>> {
+        let cache: &PackageCache = ctx.data().map_err(|_| {
+            graphql_error(
+                code::INTERNAL_SERVER_ERROR,
+                "Unable to fetch Package Cache.",
+            )
+        })?;
+        let (struct_tag, move_struct) =
+            deserialize_move_struct(&self.stored_object.serialized_object, cache)
+                .await
+                .extend()?;
+
+        // Get TypeTag of the DynamicField name from StructTag of the MoveStruct
+        let dynamic_field_type = self.dynamic_field_type()?;
+        let type_tag = DynamicFieldInfo::try_extract_field_name(&struct_tag, &dynamic_field_type)
+            .map_err(|e| Error::Internal(e.to_string()).extend())?;
+        let undecorated: Option<value::MoveValue>;
+
+        let name_move_value = extract_field_from_move_struct(move_struct, "name")?;
+
+        if self.is_dynamic_object() {
+            let inner_name_move_value = match name_move_value {
+                value::MoveValue::Struct(inner_struct) => {
+                    extract_field_from_move_struct(inner_struct, "name")
+                }
+                _ => Err(Error::Internal("Unexpected wrapper type".to_string()).extend()),
+            }?;
+            undecorated = Some(inner_name_move_value.undecorate());
+        } else {
+            undecorated = Some(name_move_value.undecorate());
+        }
+
+        let bcs = if let Some(ref undec) = undecorated {
+            bcs::to_bytes(undec)
+                .map_err(|e| Error::Internal(format!("Failed to serialize object: {e}")).extend())
+        } else {
+            Err(Error::Internal("No value to serialize".to_string()).extend())
+        }?;
+
+        Ok(Some(MoveValue::new(
+            type_tag.to_canonical_string(true),
+            Base64::from(bcs),
+        )))
+    }
+
+    async fn value(&self, ctx: &Context<'_>) -> Result<Option<DynamicFieldValue>> {
+        if self.is_dynamic_object() {
+            let address = self
+                .stored_object
+                .df_object_id
+                .as_ref()
+                .map(SuiAddress::from_bytes)
+                .transpose()?;
+
+            match address {
+                Some(address) => {
+                    let obj = ctx
+                        .data_unchecked::<PgManager>()
+                        .fetch_move_obj(address, None)
+                        .await
+                        .extend()?;
+                    Ok(obj.map(DynamicFieldValue::MoveObject))
+                }
+                None => Ok(None),
```

Comment:
Should this be an error case? When is it okay for a row in the `objects` table to indicate that it's a dynamic object field but not have a `df_object_id` entry?